### PR TITLE
Rec: stack protector for mthread stacks

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Place a `PROT_NONE` guard page at top and bottom of stack. Original work by @rgacogne with some tweaks by me.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [X] included documentation (including possible behaviour changes)
- [X] documented the code
- [ ] added or modified regression test(s)
- [X] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/test-mtasker.cc [hunk 1]
```diff
@@ -40,6 +41,43 @@ BOOST_AUTO_TEST_CASE(test_Simple)
   BOOST_CHECK_EQUAL(g_result, o);
 }
 
+static const size_t stackSize = 8 * 1024;
+static const size_t headroom = 256; // Decrease to hit stackoverflow
+
+static void doAlmostStackoverflow(void* arg)
+{
+  auto* mt = reinterpret_cast<MTasker<>*>(arg);
+  int localvar[stackSize / sizeof(int) - headroom]; // expermimentally derived headroom
```
Comment (rgacogne) [inline pdns/recursordist/test-mtasker.cc:50]:
```suggestion
  int localvar[stackSize / sizeof(int) - headroom]; // experimentally derived headroom
```

File: pdns/recursordist/test-mtasker.cc [hunk 2]
```diff
@@ -40,6 +41,43 @@ BOOST_AUTO_TEST_CASE(test_Simple)
   BOOST_CHECK_EQUAL(g_result, o);
 }
 
+static const size_t stackSize = 8 * 1024;
+static const size_t headroom = 256; // Decrease to hit stackoverflow
```
Comment (rgacogne) [inline pdns/recursordist/test-mtasker.cc:45]: Right the required headrom is a lot lower for me (112), but I don't quite see how to determine that automatically without too much trouble.

**Review by rgacogne — APPROVED** [review]

**Comment by rgacogne** [comment]
Ah, the head-room is not enough with TSAN, apparently..

File: pdns/recursordist/test-mtasker.cc (hunk 2 above)
Comment (omoerbeek) [reply pdns/recursordist/test-mtasker.cc:45]: 128 did not work on arm64, so i picked the next power of two

**Comment by omoerbeek** [comment]
Looks like ASAN+UBSAN needs even more

**Pull request merged**